File: bots/csharpClient/Bots/Obstacle.cs
using System.Text.Json.Serialization;

namespace CsClient.Bots
{
    public class Obstacle
    {
        [JsonPropertyName("type")]
        public string Type { get; set; }

        [JsonPropertyName("origin_x")]
        public float OriginX { get; set; }

        [JsonPropertyName("origin_y")]
        public float OriginY { get; set; }

        [JsonPropertyName("height")]
        public int Height { get; set; }

        [JsonPropertyName("width")]
        public int Width { get; set; }

        [JsonPropertyName("close_area_height")]
        public int CloseAreaHeight { get; set; }

        [JsonPropertyName("close_area_width")]
        public int CloseAreaWidth { get; set; }

        [JsonConstructor]
        public Obstacle(string type, float originX, float originY, int height, int width, int closeAreaHeight, int closeAreaWidth)
        {
            Type = type;
            OriginX = originX;
            OriginY = originY;
            Height = height;
            Width = width;
            CloseAreaHeight = closeAreaHeight;
            CloseAreaWidth = closeAreaWidth;
        }
    }
}

File: bots/kotlinClient/src/main/kotlin/cb/flappybuddy/game/Obstacle.kt
package cb.flappybuddy.game

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class Obstacle(
    @SerialName("type") val type: String,
    @SerialName("origin_x") var originX: Double,
    @SerialName("origin_y") var originY: Double,
    @SerialName("height") val height: Double,
    @SerialName("width") val width: Double,
    @SerialName("close_area_height") val closeAreaHeight: Double,
    @SerialName("close_area_width") val closeAreaWidth: Double
)
